binding/format/protobuf/v2/internal/pb/gen.go

```diff
@@ -0,0 +1,3 @@
+package pb
+
+//go:generate protoc --go_out=. cloudevent.proto
```

Comment:
I pointed out `cloudevent.pb.go` that you've used a development build of protoc-gen-go. We need to make sure developers running `go generate` have the correct dependencies.

One way I have solved this problem in the past is to have an additional `tools.go` file with a build tag so that it is not built with the source, but dependencies are tracked in go.mod, see this [comment](https://github.com/golang/go/issues/25922#issuecomment-412992431) for more background:

```go
// +build tools

package pb

// import packages required to run protoc commands, the versions will be tracked in go.mod
import (
	_ "google.golang.org/protobuf/cmd/protoc-gen-go"
)
```

Then you have a shell script, `generate.sh`:

```sh
# puts the protoc-gen-go executable in $GOBIN
go install google.golang.org/protobuf/cmd/protoc-gen-go
protoc --go_out=. cloudevent.proto
```

And update this `gen.go` file to simply be:

```go
//go:generate sh generate.sh
package pb
```

_Disclaimer: I've copied the above snippets from a codebase which I maintain, but we use vendoring so had to modify a bit, not sure whether it's 100% correct._

Reply:
Just noticed, this only solves the problem of having the correct version of `protoc-gen-go` installed, developers would still need to manually install the `protoc` compiler. One to address later I think?

Reply:
I use a similar technique for other projects. One variation is to make a `tools` submodule that isolates the build dependencies from the real ones. For example, folks could `go get github.com/cloudevents/sdk-go/binding/format/protobuf/v2` without downloading the `protoc` plugin but folks with the source cloned could `cd binding/format/protobuf/v2/tools && GOBIN="/path/to/protoc/plugins" go get`.

I can set up either one but they both require that the developer have `GOBIN` on their path for it to work and protoc to be installed separately. Alternatively, I can set up a `Makefile` or script that can manage downloading those dependencies and coordinating the path in a way that doesn't use the developer's global `GOBIN`. The downside is that it's difficult to make something work across all development environments that way and particularly between windows and others.